dtale/views.py

```diff
@@ -3140,6 +3157,20 @@ def web_upload():
             "from dtale.cli.loaders.json_loader import loader_func as load_json\n\n"
             "df = load_csv(path='{url}'{proxy})"
         ).format(url=url, proxy=", '{}'".format(proxy) if proxy else "")
+    elif data_type=="excel":
+        #pd.read_excel() returns a dict of dataframes keyed by sheet name.
+        dfs = load_excel(path=url, proxy=proxy)
+        dfs=list(dfs.values())
+        for i in range(len(dfs)):
+            df=dfs[i]            
+            if i==len(dfs)-1:
+                return load_new_data(df, (
+                        "from dtale.cli.loaders.excel_loader import loader_func as load_excel\n\n"
+                        "df = load_excel(path='{url}'{proxy},engine='openpyxl')"
+                        ).format(url=url, proxy=", '{}'".format(proxy) if proxy else ""))
+            else:
+                load_new_data(df,"")
```

Comment:
If you make the updates to the excel CLI loader I listed then we should be able to make these updates:

```suggestion
    elif data_type=="excel":
        df = load_excel(path=url, proxy=proxy)
        startup_code = (
            "from dtale.cli.loaders.excel_loader import loader_func as load_excel\n\n"
            "df = load_excel(path='{url}'{proxy})"
        ).format(url=url, proxy=", '{}'".format(proxy) if proxy else "")
```